src/restful/controllers/applicationsControllers.js

```diff
@@ -0,0 +1,189 @@
+import dotenv from "dotenv";
+import Util from "../../helper/utils";
+import { v4 as uuidv4 } from 'uuid';
+const admin = require("firebase-admin");
+
+dotenv.config();
+
+const util = new Util();
+
+class ApplicationController {
+  static async getAllApplications(req, res) {
+    const db = admin.firestore();
+    try {
+      const applicationsData = [];
+  
+      // Fetch all documents from the "applications" collection
+      const querySnapshot = await db.collection('applications').get();
+  
+      // Iterate over each document
+      querySnapshot.forEach(doc => {
+        const data = doc.data();
+        applicationsData.push(data);
+      });
+  
+      if (applicationsData.length > 0) {
+        util.statusCode = 200;
+        util.message = applicationsData;
+        return util.send(res);
+      } else {
+        util.statusCode = 404;
+        util.message = "Not found";
+        return util.send(res);
+      }
+    } catch (error) {
+      console.log(error);
+      util.statusCode = 500;
+      util.message = error.message || "Server error";
+      return util.send(res);
+    }
+  }
+
+  static async getApplicationById(req, res) {
+    const db = admin.firestore();
+    const { application_id } = req.params;
+    try {
+      const applicationSnapshot = await db.collection('applications').doc(application_id).get();
+      if (applicationSnapshot.exists) {
+        const applicationData = applicationSnapshot.data();
+        util.statusCode = 200;
+        util.message = applicationData;
+        return util.send(res);
+      } else {
+        util.statusCode = 404;
+        util.message = "Application not found";
+        return util.send(res);
+      }
+    } catch (error) {
+      console.log(error);
+      util.statusCode = 500;
+      util.message = error.message || "Server error";
+      return util.send(res);
+    }
+  }
+
+  static async createApplication(req, res) {
+    const db = admin.firestore();
+    const { user_id, opportunity_id, status } = req.body;
```

Comment:
The req body should include a `proposal` key so that we can send the application text data